src/Devito.jl

```diff
@@ -256,7 +265,7 @@ end
 PyCall.PyObject(x::Grid) = x.o
 
 Base.size(grid::Grid{T,N}) where {T,N} = reverse((grid.o.shape)::NTuple{N,Int})
-extent(grid::Grid{T,N}) where {T,N} = reverse((grid.o.extent)::NTuple{N,T})
+extent(grid::Grid{T,N}) where {T,N} = reverse((convert(NTuple{N,T},grid.o.extent))::NTuple{N,T})
```

Comment:
I'm not sure I understand this change.  Was the previous version failing?

Reply:
yes, the previous version failed unit tests because extent returns a Float64 naturally.
Part of associating dtype {T} with the grid is that returning a tuple of a different type causes this to fail